### ISSUE TITLE
Still usable?

### ISSUE THREAD

**Issue by debugged-hosting** [pducharme/UniFi-Video-Controller#201]
Is UniFi Video Docker still usable? I have clients that do not want video footage stored on-site and this would be perfect so I could pair the cameras to this controller that runs off-site.

**Comment by APz** [comment]
Yes. I have couple of sites that were not practical to migrate to Unifi Protect and to this day they run the .13 version of Unifi Video. Since the Unifi Video portal is killed, you need to use VPN or port forwards (highly not recommended) to access it outside. Other than that, it works and has the limitations it always had, like eternal beta timeline and so on.

**Comment by adminy** [comment]
I got an AI 360 camera. It is version is UVC.v4.57.35.67.M_d45274d.221013.0614 and it does not work with the container. There is gotta be some newer version alternative out there. Why did they stop producing new deb files? could they have at least open sourced it so that the product isn't dead without their ultra expensive server?

**Comment by vuhuy** [comment]
The original author disbanded the project for good reasons, as UniFi Video is now EOL. However, the images at Docker Hub are outdated vulnerable for the log4j exploit. An unreleased fix is available from the master branch but it won't build due to broken dependencies.

I made a [fork](https://github.com/vuhuy/docker-unifi-video) with some changes and fixes. I'm not planning to submit a pull request since UniFi Video and this repostory are dead.

- Updated container base image. Changed from phusion/baseimage:0.11 to phusion/baseimage:bionic-1.0.0-amd64.
- Rework correct port exposure. Original had to many unnecessary ports exposed.
- Rework available environment variables. Added and renamed a few options to configure the container using Docker run or Compose.
- Rework log4j mitigation. Changed mitigation method from disguising log4j v2.17.0 as v2.1.0 to removing the JndiLookup class from the classpath of the included log4j v2.1.0 in UniFi Video.
- Rework UniFi Video and dependencies installation script. This also fixes the broken Docker build on the current main branch of the original repository.
- Rework host TMPFS fix. The TMPFS should now be populated before the UniFi Video service starts. Should work for container restarts too.
- Updated README file. More examples and documentation.
- Updated Docker Compose YAML. Is now up to date with all changes done.

Works fine for Docker on Ubuntu 22.04 LTS. Should still work for k8s too but I have not tested this.
